### ISSUE TITLE
Changing GM minutes is confusing

### ISSUE THREAD

**Issue by thaliawww-service** [thaliawww/concrexit#882]
In GitLab by ahoeks on Jun 4, 2019, 18:05

### One-sentence description

It looks like minutes is a required field if you set the 'Members only' checkbox to true.

### Current behaviour / Reproducing the bug

When I want to make a General Meeting 'members only' I get the error that Minutes are a required field.

### Expected behaviour

I expect to be able to make changes to the information without having to upload minutes.

### Suggested solution

Set the `extra` field of the InlineModelAdmin to 0. https://docs.djangoproject.com/en/2.2/ref/contrib/admin/#django.contrib.admin.InlineModelAdmin.extra

This way the default minutes form will be hidden and a user will be required to add it manually.

**Comment by thaliawww-service** [comment]
In GitLab by @se-bastiaan on Jun 4, 2019, 20:19

*TL;DR: If you do not change the fields in the minutes part of the admin you can still save the GM. GM's are always public, documents are not.*

There are two types of situations you can be in:

1. Files have been uploaded and minutes are available.

![Screenshot_2019-06-04_at_20.09.58](https://gitlab.science.ru.nl//thalia/concrexit/uploads/fddcaed865378a77fd720bdbc8c478d6/Screenshot_2019-06-04_at_20.09.58.png)

Changing this checkbox will cause an error because of #882. The file has been uploaded and the GM cannot be modified due to the file validation error. This in turn causes the GM to not be editable, which is expected behaviour since the file should be correct. For the minutes to be correct. It would be possible to change the GM if the Minutes are deleted, which is possible by using the delete checkbox in the upper-right of the minutes form.

![Screenshot_2019-06-04_at_20.12.01](https://gitlab.science.ru.nl//thalia/concrexit/uploads/9c82ddaf9d9c14cfa8f1888603ded4eb/Screenshot_2019-06-04_at_20.12.01.png)

2. There are no minutes yet.

![Screenshot_2019-06-04_at_20.10.36](https://gitlab.science.ru.nl//thalia/concrexit/uploads/85a3ad516f7721b97ccb397f5a154afb/Screenshot_2019-06-04_at_20.10.36.png)

This is the situation where the form for the minutes does exist, but the actual document representing the minutes has not been created yet. A good indicator for this situation is checking if the previously mentioned checkbox is available.

Changing the checkbox here would create the document representation of the minutes to save this value, which means that the file fields of the document become required. A document representation cannot exist without files. The minutes file is not a part of the general meeting itself, so if the object does not exist you can still save the GM. But requiring login for a document that does not exist is not possible. The 'members only' checkbox is a part of the minutes, the GM date itself is always public.

PS. Please never change the default issue priority.

**Comment by thaliawww-service** [comment]
In GitLab by @se-bastiaan on Jun 4, 2019, 20:19

closed

**Comment by thaliawww-service** [comment]
In GitLab by @se-bastiaan on Jun 4, 2019, 20:20

reopened

**Comment by thaliawww-service** [comment]
In GitLab by @se-bastiaan on Jun 20, 2019, 16:59

assigned to @se-bastiaan

**Comment by thaliawww-service** [comment]
In GitLab by @pingiun on Jun 26, 2019, 20:13

closed via merge request !1297

**Comment by thaliawww-service** [comment]
In GitLab by @pingiun on Jun 26, 2019, 20:13

mentioned in commit b1e71d2e4942a631d23d6882f62d04a9ec66943c
